Samples/Catalog/Sources/AlertController.swift

```diff
@@ -19,12 +19,94 @@
 import Foundation
 import UIKit
 
+extension UIAlertAction {
+  
+  /// Action types most commonly used
+  public enum actionType{
+    
+    ///Ok Option
+    case ok
+    
+    /// Default Cancel Option
+    case cancel
+    
+    /// Destructive action with custom title
+    case destructive(String)
+    
+    /// Custom action with title and style
+    case custom(String,UIAlertActionStyle)
+    
+    /**
+     Creates the action instance for UIAlertController
+     - parameter handler: Call Back function
+     - returns UIAlertAction Instance
+     */
+    public func action(handler:((String) -> Void)? = nil) -> UIAlertAction {
+      
+      //Default value
+      var actionStyle = UIAlertActionStyle.default
+      var title = ""
+      
+      // Action configuration based on the action type
+      switch self {
+        
+      case .cancel:
+        actionStyle = .cancel
+        title = "Cancel"
+        
+      case .destructive(let optionTitle):
+        title = optionTitle
+        actionStyle = .destructive
+        
+      case .custom(let optionTitle, let style):
+        title = optionTitle
+        actionStyle = style
+        
+      default:
+        title = "Ok"
+      }
+      
+      //Creating UIAlertAction instance
+      let action = UIAlertAction(title:title,style:actionStyle, handler: {(nativeAction) in
+        if (handler != nil){
+          handler!(title)
+        }
+      })
+      
+      return action
+    }
+  }
+}
+
 extension UIAlertController {
-  convenience init(title: String, message: String) {
-    self.init(title: title, message: message, preferredStyle: .alert)
-    let action = UIAlertAction(title: NSLocalizedString("OK", comment: "OK action"),
-                               style: .default,
-                               handler: nil)
-    addAction(action)
+  
+  /**
+   Creates the alert view controller using the actions specified
+   
+   - parameter title:  Title of the alert.
+   - parameter message: Alert message body.
+   - parameter actions: Variable numbre of actions as an Array of actionType values.
+   - parameter style: UIAlertControllerStyle enum value
+   - parameter handler: Handler block/closure for the clicked option.
+   
+   -     let alert = UIAlertController.init(title:"Confirm",message:"Do you want to quit sign up and login with other account ?",actions:.custom("Yes,.default),.custom("Cancel",.destructive)){ (buttonTitle) in
+   if buttonTitle == "Yes"{
+   print("Yes Clicked")
+   }
+   }
+   
+   self.present(alert, animated: true, completion: nil)
```

Comment:
Not sure if this line is relevant, since it suggests presentation from inside a `UIViewController`. I suggest removing it.